### PR TITLE
Add ability to clean field names for set_field and set_fields pipeline functions

### PR DESCRIPTION
**Pull request by danotorrey**
## Description

Add optional parameters to the following pipeline functions that can be used to replace invalid field name characters with underscores. Both default to `false`.

* `set_field` pipeline function: New parameter: `clean_field`
* `set_fields` pipeline function: New parameter: `clean_fields`

**`set_field` example**

```
rule "set_field"
when true
then
  set_field(field: "has space chars", value: "value", clean_field: true);
end
```

In this example, the field `has_space_chars` will be added to the message with the value `value`.

**`set_fields` example**
Given a message with a `json_map` field containing the following value:

```
{
  "key_1": "value 1",
  "has space chars": "value"
}
```

And the following rule:
```
rule "set_fields"
when true
then
  let cleanFieldValue = to_map(parse_json(to_string($message.json_map)));
  set_fields(fields: cleanFieldValue, clean_fields: true);
end

```

In this example, the same field with a spaces in it will be successfully added to the message (along with the other field `key_1`: `value`).

## Motivation and Context

When attempting to use a JSON map to set fields on a message, any fields that contain invalid characters (spaces, for example) will not be added to the message. This new optional parameter provides a backwards-compatible and convenient  way to automatically clean up invalid field names and ensure that all fields in the map will be set on the message. 

Closes https://github.com/Graylog2/graylog-plugin-enterprise/issues/5794
Closes https://github.com/Graylog2/graylog2-server/issues/7137

## Screenshots
image
image

## How Has This Been Tested?

Unit and manual tests.

You can send a message with a JSON map in using the above `set_fields` example using this GELF HTTP example:
```
curl -XPOST -v http://127.0.0.1:12201/gelf -p0 \
-d $'{"short_message":"Bulk message 1", "host":"example.org", "facility":"test", "_json_map":"{ \\"key_1\\": \\"value 1\\", \\"k 6\\": \\"value\\" }"}'
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by ryan-carroll-graylog — APPROVED** [review]
Looks good and tests successfully!

File: graylog2-server/src/test/java/org/graylog2/plugin/MessageTest.java
Comment (kingzacko1) [inline graylog2-server/src/test/java/org/graylog2/plugin/MessageTest.java:398]: nit: spacing between parameters. There are some other places in this file that could use this same thing so if you could run a Code > Reformat Code it should clean them all up.

**Review by kingzacko1 — APPROVED** [review]
LGTM!

**Comment by danotorrey** [comment]
Thanks for the reviews @ryan-carroll-graylog @kingzacko1! I will clean up that formatting issue in a few!

Also, it turns out that an index of pipeline functions is present in the docs, so I submitted an Asana ticket to add this param to the existing sections: 
https://go2docs.graylog.org/5-0/making_sense_of_your_log_data/functions_descriptions.html?Highlight=functions%20descriptions#setfield
https://go2docs.graylog.org/5-0/making_sense_of_your_log_data/functions_descriptions.html?Highlight=functions%20descriptions#setfields

File: graylog2-server/src/test/java/org/graylog2/plugin/MessageTest.java
Comment (danotorrey) [reply graylog2-server/src/test/java/org/graylog2/plugin/MessageTest.java]: Thanks for catching this! I went ahead and formatted both test files, so there are some more formatting cleanup included in the commit.

**Comment by danotorrey** [comment]
I will plan to wait until tomorrow to merge this incase anyone else has feedback.

**Pull request merged**
